app/controllers/journeys/nics/Class4NonDivingExemptSingleBusinessController.scala

```diff
@@ -0,0 +1,91 @@
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.journeys.nics.Class4NonDivingExemptYesNoView
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.Future
+
+@Singleton
+class Class4NonDivingExemptSingleBusinessController @Inject() (override val messagesApi: MessagesApi,
+                                                               val controllerComponents: MessagesControllerComponents,
+                                                               identify: IdentifierAction,
+                                                               getData: DataRetrievalAction,
+                                                               requireData: DataRequiredAction,
+                                                               formProvider: BooleanFormProvider,
+                                                               service: SelfEmploymentService,
+                                                               view: Class4NonDivingExemptYesNoView)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val page = Class4NonDivingExemptSingleBusinessPage
+
+  def onPageLoad(taxYear: TaxYear, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
+    page.remainingBusinesses(request.userAnswers) match {
+
+      case List(Business(_, tradingName)) =>
+        val filledForm = request.userAnswers
+          .get(page, nationalInsuranceContributions)
+          .flatMap { businessIds =>
+            businessIds.headOption.collect {
+              case id if id == classFourNoneExempt => false
+              case _                               => true
+            }
+          }
+          .foldLeft(formProvider(page, request.userType, parameters = List(s"${tradingName.value}."))) { (form, value) =>
+            form.fill(value)
+          }
+
+        Ok(view(filledForm, taxYear, request.userType, mode, tradingName.value))
+      case invalidList => InternalServerError(s"Controller called with invalid remainingBusinesses list ${invalidList.mkString("[", ",", "]")}")
+    }
+  }
+
+  def onSubmit(taxYear: TaxYear, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    def handleError(formWithErrors: Form[_]): Result =
+      BadRequest(view(formWithErrors, taxYear, request.userType, mode, page.remainingBusinesses(request.userAnswers).head.tradingName.value))
+
+    def handleSuccess(answer: Boolean): Future[Result] =
+      if (answer) {
+        service.persistAnswerAndRedirect(
+          page,
+          nationalInsuranceContributions,
+          request,
+          page.remainingBusinesses(request.userAnswers).map(_.businessId),
+          taxYear,
+          mode)
+      } else {
+        service.persistAnswerAndRedirect(page, nationalInsuranceContributions, request, List(classFourNoneExempt), taxYear, mode)
+      }
```

Comment:
Optional change. Would this be more clear?
` def handleSuccess(answer: Boolean): Future[Result] = {
      val submissionAnswer = if (answer) page.remainingBusinesses(request.userAnswers).map(_.businessId) else List(classFourNoneExempt)
      service.persistAnswerAndRedirect(page, nationalInsuranceContributions, request, submissionAnswer, taxYear, mode)
    }`